### PR TITLE
build: fix pkg-config file generation

### PR DESCRIPTION
**Pull request by Tachi107**
- Instead of hardcoding "lib" and "include" in `libdir` and `includedir`, use the values from [`GNUInstallDirs`](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html).

- Use `PROJECT_DESCRIPTION` and `PROJECT_HOMEPAGE_URL` instead of their
`CMAKE_` conterparts to fix pkg-config generation when rocksdb is not the top-level project (see [`project()`](https://cmake.org/cmake/help/latest/command/project.html)).

- Drop explicit `CMAKE_CURRENT_SOURCE_DIR` and `CMAKE_CURRENT_BINARY_DIR` in [`configure_file()`](https://cmake.org/cmake/help/latest/command/configure_file.html) as that's implied by default (and quite intuitive).

See #9945
CC: @trynity

### CODE REVIEW

**Comment by trynity** [comment]
Ah, thanks for the cleanup! First time trying to generate PC files from CMake, so was bound to get a few things not the cleanest.

**Comment by Tachi107** [comment]
Converted to draft as it a few things don't work as expected

**Comment by tristan957** [comment]
Some of us recently put in the effort to add pkg-config support to HdrHistogram_c which is also CMake-based. From the looks of it, your pkg-config file has a pitfall which was corrected in the HdrHistogram_c repo in this commit: https://github.com/HdrHistogram/HdrHistogram_c/commit/b083efd27a51150201821ef6f18e0eba1f4b59f1. You may want to do a similar fix.

**Comment by Tachi107** [comment]
> From the looks of it, your pkg-config file has a pitfall which was corrected in the HdrHistogram_c repo in this commit: [HdrHistogram/HdrHistogram_c@b083efd](https://github.com/HdrHistogram/HdrHistogram_c/commit/b083efd27a51150201821ef6f18e0eba1f4b59f1).

Hi, thanks for the feedback :)

@jtojnar's JoinPaths.cmake says that Windows is not supported (even though by looking at the code I don't understand why). Fixing generation of pkg-config files when absolute directories are specified at the cost of breaking Windows support doesn't look like a good trade off to me

**Comment by tristan957** [comment]
you can just special case Windows to do what you are currently doing.

**Comment by jtojnar** [comment]
> @jtojnar's JoinPaths.cmake says that Windows is not supported (even though by looking at the code I don't understand why).

I wrote a bit about it here: https://github.com/jtojnar/cmake-snips/issues/4

TLDR: Windows has two more path types (absolute paths without drive letter and relative paths with drive letter) and I have no idea how it behaves there. Then there is using `/` as directory separator but that might not actually be an issue.

If you are able to raise minimum CMake version to 3.20, you can replace the custom module with `cmake_path(APPEND …)`, see https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files.

**Comment by tristan957** [comment]
Shame this got pulled in even though it is broken, imo.

**Review by ajkr — APPROVED** [review]
LGTM
